phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/HighAvailabilityGroup.java

```diff
@@ -785,26 +851,49 @@ public String getUrl2() {
             return urls.getSecond();
         }
 
-        public String getJDBCUrl(String zkUrl) {
-            Preconditions.checkArgument(zkUrl.equals(getUrl1()) || zkUrl.equals(getUrl2()),
+
+        //It applies only the current thread context's URL Info as we have 1:n mapping between
+        //HAURLInfo -> HAGroupInfo.
```

Comment:
nit: Isn't the 1:n mapping from HAGroupInfo -> HAURLInfo?

Reply:
Forgot to remove comment it was with other approach, will update this